### PR TITLE
WikiPage-V2

### PR DESCRIPTION
**Pull request by heidiso**
Made code review changes for the DAO (cleanup) and TranslationUtils (more tests)
Added manager, service, and controller

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+		Map<String, FileHandle> nameToHandleMap = buildFileNameMap(wikiPage);
+		// Validate that the user can assign all file handles
+		for(FileHandle handle: nameToHandleMap.values()){
+			// the user must have access to the raw FileHandle to assign it to an object.
+			if(!authorizationManager.canAccessRawFileHandleByCreator(user, handle.getCreatedBy())){
+				throw new UnauthorizedException(String.format(USER_IS_NOT_AUTHORIZED_FILE_HANDLE_TEMPLATE, handle.getId()));
+			}
+		}
+		// pass to the DAO
+		return wikiPageDao.create(wikiPage, nameToHandleMap, objectId, objectType);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java:104]: should also tell the dao what the filehandles to add are.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,308 @@
+package org.sagebionetworks.repo.manager.wiki;
+
+import java.util.ArrayList;
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java:3]: Wiki.title should also be part of the version.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java:6]: double check each switch is covered by a test.

File: services/repository/src/main/java/org/sagebionetworks/repo/web/UrlHelpers.java [hunk 4]
```diff
@@ -583,6 +583,37 @@
 	public static final String EVALUATION_WIKI_ID_ATTCHMENT_HANDLE =EVALUATION_OWNER_ID + WIKI_WITH_ID+ATTACHMENT_HANDLES;
 	public static final String EVALUATION_WIKI_ID_ATTCHMENT_FILE =EVALUATION_OWNER_ID + WIKI_WITH_ID+ATTACHMENT;
 	public static final String EVALUATION_WIKI_ID_ATTCHMENT_FILE_PREVIEW =EVALUATION_OWNER_ID + WIKI_WITH_ID+ATTACHMENT_PREVIEW;	
+
+	// V2 Wiki URL
+	public static final String WIKI_V2 = "/wiki2";
+	public static final String WIKI_HEADER_TREE_V2 = "/wikiheadertree2";
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/UrlHelpers.java:589]: make sure there is only one 2 in the URL.

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/V2WikiController.java [hunk 5]
```diff
@@ -0,0 +1,947 @@
...
+			@RequestParam(value = AuthorizationConstants.USER_ID_PARAM, required = false) String userId,
+			@PathVariable String ownerId, @PathVariable String wikiId)
+			throws DatastoreException, NotFoundException {
+		return serviceProvider.getV2WikiService().getWikiPage(userId,
+				new WikiPageKey(ownerId, ObjectType.EVALUATION, wikiId));
+	}
+
+	// Update methods.
+	/**
+	 * Update a specific WikiPage of an Entity.
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/V2WikiController.java:268]: add a comment that each update creates a new version

File: services/repository/src/test/java/org/sagebionetworks/repo/web/controller/V2WikiControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = { "classpath:test-context.xml" })
+public class V2WikiControllerTest {
+	@Autowired
+	private EntityServletTestHelper entityServletHelper;
+	@Autowired
```
Comment (john-hill) [inline services/repository/src/test/java/org/sagebionetworks/repo/web/controller/V2WikiControllerTest.java:46]: add paginated tests where limit and offset are null
